example/spec/helpers/kinesisHelpers.js

```diff
@@ -213,41 +214,63 @@ async function putRecordOnStream(streamName, record) {
   }).promise();
 }
 
-
 /**
- * Wait for test stepfunction execution to exist.
+ * Wait for a certain number of test stepfunction executions to exist.
  *
  * @param {string} recordIdentifier - random string identifying correct execution for test
+ * @param {string} workflowName - name of the workflow to wait for
  * @param {integer} maxWaitTimeSecs - maximum time to wait for the correct execution in seconds
+ * @param {integer} numExecutions - The number of executions to wait for
  * @param {string} firstStep - The name of the first step of the workflow, used to query if the workflow has started.
  * @returns {Object} - {executionArn: <arn>, status: <status>}
  * @throws {Error} - any AWS error, re-thrown from AWS execution or 'Workflow Never Started'.
  */
-async function waitForTestSf(recordIdentifier, maxWaitTimeSecs, firstStep = 'SfSnsReport') {
+async function waitForAllTestSf(recordIdentifier, workflowName, maxWaitTimeSecs, numExecutions, firstStep = 'SfSnsReport') {
   let timeWaitedSecs = 0;
-  let workflowExecution;
+  const workflowExecutions = [];
+  const startTime = moment();
 
   /* eslint-disable no-await-in-loop */
-  while (timeWaitedSecs < maxWaitTimeSecs && workflowExecution === undefined) {
+  while (timeWaitedSecs < maxWaitTimeSecs && workflowExecutions.length < numExecutions) {
     await sleep(waitPeriodMs);
-    timeWaitedSecs += (waitPeriodMs / 1000);
-    const executions = await getExecutions('KinesisTriggerTest', testConfig.stackName, testConfig.bucket, maxExecutionResults);
+    timeWaitedSecs = (moment.duration(moment().diff(startTime)).asSeconds());
```

Comment:
We had a bug here where basing the time on the sleep time was way off. This makes it difficult to set the proper timeouts.